src/main/java/games/strategy/util/Match.java

```diff
@@ -172,4 +185,79 @@ public boolean match(final T value) {
       }
     };
   }
+
+  /**
+   * Creates a new match whose condition is satisfied if the test object matches any of the specified conditions.
+   *
+   * @param matches An array of matches; must not be {@code null}.
+   *
+   * @return A new match; never {@code null}.
+   */
+  @SafeVarargs
+  public static <T> Match<T> any(final Match<T>... matches) {
+    checkNotNull(matches);
+
+    return any(Arrays.asList(matches));
```

Comment:
I see, I just messed around with The CompoisteMatchAnd class and this is a generic type mess!
I'm aiming to replace those classes with the stream API